core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -478,6 +478,21 @@ public static double maxRadialDistanceMeters(final double centerLat, final doubl
       return SloppyMath.haversinMeters(centerLat, centerLon, centerLat, (MAX_LON + centerLon) % 360);
     }
 
+    /** Return the distance (in meters) between 2 lat,lon geo points using the haversine method implemented by lucene */
+    public static double arcDistance(double lat1, double lon1, double lat2, double lon2, DistanceUnit unit) {
```

Comment:
I don't think we need the flexibility to take distance unit here. That is an unnecessary extra step. If a user wants to convert from meters, they can. But they don't need our internal utility to do it. Having it also disagrees with the javadoc, which states meters are returned.